test/accounts.js

```diff
@@ -1,263 +1,195 @@
-var BN = require("bn.js");
-var Web3 = require("web3");
-var Ganache = require(process.env.TEST_BUILD
-  ? "../build/ganache.core." + process.env.TEST_BUILD + ".js"
-  : "../index.js");
-var assert = require("assert");
+const assert = require("assert");
+const initializeTestProvider = require("./helpers/web3/initializeTestProvider");
 
-describe("Accounts", function() {
-  var expectedAddress = "0x604a95C9165Bc95aE016a5299dd7d400dDDBEa9A";
-  var mnemonic = "into trim cross then helmet popular suit hammer cart shrug oval student";
+describe("Accounts", async() => {
+  const expectedAddress = "0x604a95c9165bc95ae016a5299dd7d400dddbea9a";
+  const badAddress = "0x1234567890123456789012345678901234567890";
+  const mnemonic = "into trim cross then helmet popular suit hammer cart shrug oval student";
 
-  it("should respect the BIP99 mnemonic", function(done) {
-    var web3 = new Web3();
-    web3.setProvider(
-      Ganache.provider({
-        mnemonic: mnemonic
-      })
-    );
+  it("should respect the BIP99 mnemonic", async() => {
+    const options = { mnemonic };
+    const { accounts } = await initializeTestProvider(options);
 
-    web3.eth.getAccounts(function(err, accounts) {
-      if (err) {
-        return done(err);
-      }
-
-      assert(accounts[0].toLowerCase(), expectedAddress.toLowerCase());
-      done();
-    });
+    assert.strictEqual(accounts[0].toLowerCase(), expectedAddress);
   });
 
-  it("should lock all accounts when specified", function(done) {
-    var web3 = new Web3();
-    web3.setProvider(
-      Ganache.provider({
-        mnemonic: mnemonic,
-        secure: true
-      })
-    );
-
-    web3.eth.sendTransaction(
-      {
-        from: expectedAddress,
-        to: "0x1234567890123456789012345678901234567890", // doesn't need to exist
-        value: web3.utils.toWei(new BN(1), "ether"),
-        gasLimit: 90000
-      },
-      function(err) {
-        if (!err) {
-          return done(
-            new Error("We expected the account to be locked, which should throw an error when sending a transaction")
-          );
-        }
-        assert.strictEqual(err.message, "signer account is locked");
-        done();
+  it("should lock all accounts when specified", async() => {
+    const options = {
+      mnemonic,
+      secure: true
+    };
+    const { accounts, web3 } = await initializeTestProvider(options);
+
+    accounts.forEach(async(account) => {
+      try {
```

Comment:
Maybe we can do something like https://nodejs.org/api/assert.html#assert_assert_rejects_asyncfn_error_message instead of the try/catch (anti-)pattern?

Reply:
That can be done.

Reply:
`assert.rejects` was added to v10 and breaks when ran with `carbon` in Travis CI. `try/catch` appears to be the cleanest approach. I'm open to other suggestions.

Reply:
It turns out that adding an `async` wrapper to the executing function did the trick.
```
      await assert.rejects(
        async() => {
          await web3.eth.sendTransaction({
            from: account,
            to: badAddress,
            value: web3.utils.toWei("1", "ether"),
            gasLimit: 90000
          });
        },
        /signer account is locked/,
        "should not be able to unlock the count"
      );
```